### PR TITLE
Enable pytest failures on warnings from cudf

### PR DESCRIPTION
**Pull request by mroeschke**
In 24.04, cudf issues a FutureWarning when using positional indexing with `__getitem__`. The change here is to use `iloc` instead.

Additionally the test suite will now fail when a warning comes from cudf as apart of https://github.com/rapidsai/build-planning/issues/26

### CODE REVIEW

File: python/cugraph/pytest.ini
Comment (acostadon) [inline python/cugraph/pytest.ini:1]: License update needed

**Review by acostadon — CHANGES_REQUESTED** [review]
What is the schedule to merge this?
Is it for 24.04?

**Comment by mroeschke** [comment]
> What is the schedule to merge this? What is the schedule to merge this?
Is it for 24.04?

This is not critical for 24.04. Fine to de-prioritize over other PRs

**Review by rlratzel — APPROVED** [review]
Thank you very much!
We should have treated cudf warnings as test failures long ago.

**Comment by BradReesWork** [comment]
All - can we get this is as soon as possible.  while the warnings do not stop an analytic, is still looks bad on us to even have warrings

**Review by acostadon — APPROVED** [review]
Looks great !!!

**Comment by BradReesWork** [comment]
@mroeschke there is a merge conflict

**Pull request merged**